src/server/middleware/api/github.js

```diff
@@ -5,61 +5,113 @@ const { isProduction } = require('../../utils')
+              edges {
+                node {
+                  title
+                  createdAt
+                  url
+                  comments {
+                    totalCount
+                  }
+                }
+              }
+            }
+          }
+        }
+      `,
+      {
+        headers: {
+          authorization: `token ${process.env.GITHUB_TOKEN}`
+        }
+      }
+    )
+
+    const issues = repository.issues.edges.map(({ node }) => ({
+      title: node.title,
+      url: node.url,
+      comments: node.comments.totalCount,
+      date: node.createdAt
+    }))
+
+    const stars = repository.stargazers.totalCount
+
+    cache.set('issues', issues)
+    cache.set('stars', stars)
+
+    return { issues, stars }
+  } catch (e) {
+    return { error: e.message || e }
+  }
+}
+
+async function issues(_, res) {
   if (!process.env.GITHUB_TOKEN) {
-    res.status(200).json({ issues: [] })
+    res
+      .status(200)
+      .json({ issues: [], error: 'No GITHUB_TOKEN specified on the server!' })
   } else {
     if (cache.get('issues')) {
       if (!isProduction) console.log('Using cache for "issues"')
 
       res.status(200).json({ issues: cache.get('issues') })
     } else {
       console.log('Not using cache for "issues"')
+      const { issues } = await getFreshGithubData()
+      if (!issues) {
+        res.status(404)
+      } else {
+        res.status(200).json({ issues })
+      }
     }
+  }
+}
 
-    try {
-      const { repository } = await graphql(
-        `
-          {
-            repository(owner: "iterative", name: "dvc") {
-              issues(
-                first: 3
-                states: OPEN
-                orderBy: { field: CREATED_AT, direction: DESC }
-              ) {
-                edges {
-                  node {
-                    title
-                    createdAt
-                    url
-                    comments {
-                      totalCount
-                    }
-                  }
-                }
-              }
-            }
-          }
-        `,
-        {
-          headers: {
-            authorization: `token ${process.env.GITHUB_TOKEN}`
-          }
-        }
-      )
-
-      const issues = repository.issues.edges.map(({ node }) => ({
-        title: node.title,
-        url: node.url,
-        comments: node.comments.totalCount,
-        date: node.createdAt
-      }))
-
-      cache.set('issues', issues)
+async function getStars() {
+  const cachedValue = cache.get('stars')
+  if (cachedValue) {
+    if (!isProduction) console.log('Using cache for "stars"')
+    return cachedValue
+  } else {
+    console.log('Not using cache for "stars"')
+    const { stars } = await getFreshGithubData()
+    return stars
+  }
+}
 
-      res.status(200).json({ issues })
-    } catch (e) {
+async function stars(req, res) {
+  if (!process.env.GITHUB_TOKEN) {
+    res.status(403).json({ error: 'No GITHUB_TOKEN specified on the server!' })
```

Comment:
403 here, was 200 for issues? also the same small concern - it's better to keep impl details private (I mean error message)

Reply:
I can change it back to the previous behavior if that's what we want. I made the fail states more explicit because I was thrown off at the start of the PR when diagnosing why I got a successful result with no issues nor indication anything went wrong.

Reply:
I don't have any specific suggestions :) just some concerns - exposing impl details probably not good and errors codes probably should be consistent. Probably, we can use logs? And 403 itself sounds good if it helps to debug.

Reply:
Sure! I chose 403 forbidden because we're giving an error for the lack of an API key. That code alone with no message should be expressive enough to tell devs what's going on without actually exposing implementation details.

Reply:
👍